Authentication

Shumba250 opened the pull request:
### What does this PR do?
Authentication

### Description of Task to be completed
passport local strategy protection on the login and signup
get a token on the login
authenticate all admin private routes
authorize users/admin
### How should this be manually tested?
this can be tested with postman

finish (authentication)"

Shumba250 committed:
ch(signup): user signups

created GET API that retrieves all users
created GET API that retrieves a single user
created GET API that shows number of all users
created POST API that adds a user
created PATCH API that updates users information
created DELETE API that deletes a user

finishes (signups)

Shumba250 committed:
started validations

Shumba250 committed:
feat(validations): validations

validated POST API of Articles with Joi
validated PATCH API of Articles with Joi
validated POST API of contact messages with Joi
validated POST API of adding users with Joi
validated PATCH API of updating users information with Joi

Shumba250 committed:
feat(authenticatin): authenticate all rpoutes

passport localsrategy protection on the login and signup
get a token on the login
authenticate all admin private routes
authorize users/admin
finiish (authentication)

Shumba250 committed:
resolved conflicts

Shumba250 committed:
deleted resource file

wakaflorien reviewed (changes requested):
Good progress. just make sure you use recommended file/folder structure.

Shumba250 committed:
feat(authentication, jwt, passport):

jwt token on login
protected all admin private routes
passport on the login using local strategy and jwt stratgy
protected user routes like the likes

finishes (authentication)
